rust/src/transformations/make_stable_expr/expr_discrete_quantile_score/mod.rs

```diff
@@ -0,0 +1,208 @@
+            MakeTransformation,
+            "Quantile estimation requires non-null inputs"
+        );
+    }
+
+    fn validate<T: Number + RoundCast<f64>>(candidates: &Vec<f64>) -> Fallible<()> {
+        validate_candidates(
+            &candidates
+                .iter()
+                .cloned()
+                .map(T::round_cast)
+                .collect::<Fallible<_>>()?,
+        )
+    }
+
+    match active_series.field.dtype {
+        UInt32 => validate::<u32>(&candidates),
+        UInt64 => validate::<u64>(&candidates),
+        Int8 => validate::<i8>(&candidates),
+        Int16 => validate::<i16>(&candidates),
+        Int32 => validate::<i32>(&candidates),
+        Int64 => validate::<i64>(&candidates),
+        Float32 => validate::<f32>(&candidates),
+        Float64 => validate::<f64>(&candidates),
+        UInt8 | UInt16 => {
+            return fallible!(
+                FailedFunction,
+                "u8 and u16 not supported in the OpenDP Polars plugin. Please use u32 or u64."
+            )
+        }
+        dtype => {
+            return fallible!(
+                MakeTransformation,
+                "Expected numeric data type, found {:?}",
+                dtype
+            );
+        }
+    }?;
+
+    let margin = middle_domain.active_margin()?.clone();
+
+    let mpl = margin
+        .max_partition_length
+        .ok_or_else(|| err!(MakeTransformation, "Must know max_partition_length"))?;
+
+    let constants = score_candidates_constants::<u64>(Some(mpl as u64), alpha)?;
+    let (alpha_num, alpha_den, _) = constants.clone();
+
+    t_prior
+        >> Transformation::<_, _, PartitionDistance<MI>, Parallel<LInfDistance<_>>>::new(
+            middle_domain.clone(),
+            middle_domain,
+            Function::new_expr(move |input_expr| {
+                apply_plugin(
+                    input_expr,
+                    expr.clone(),
+                    DQScoreArgs {
+                        alpha,
+                        candidates: candidates.clone(),
+                        constants: Some(constants),
+                    },
+                )
+            }),
+            middle_metric,
+            Parallel(LInfDistance::new(false)),
+            StabilityMap::new_fallible(move |(l0, _, li)| {
+                let out_li = f64::inf_cast(score_candidates_map(
+                    alpha_num,
+                    alpha_den,
+                    margin.public_info == Some(MarginPub::Lengths),
+                )(li)?)?;
+                Ok((*l0, out_li))
+            }),
+        )?
+}
+
+/// Determine if the given expression is a discrete quantile score expression.
+///
+/// # Arguments
+/// * `expr` - The expression to check
+///
+/// # Returns
+/// If matched, ipnut expression and discrete quantile score arguments
+pub(crate) fn match_dq_score(expr: &Expr) -> Fallible<Option<(&Expr, DQScoreArgs)>> {
+    let Some((score_input, args)) = match_plugin(expr, "dq_score")? else {
```

Comment:
Similar to another comment, should occurrences of this string be replaced by constants?